### 1	

Write a program to print the numbers 2 to 10 in increments of two on the same line. The output of your program should be 2 4 6 8 10 

In [16]:
for prob in range(2, 12, 2):
    print(prob)


2
4
6
8
10


In [17]:
count = 2
end = 10
while count <= end:
    print(count)
    count += 2

2
4
6
8
10


### 2	
Write a Program to compute the sum of eight numbers. You will prompt the user for each number. Your prompt should also indicate which number is currently being accepted e.g. your prompt can be "Please enter value #5: "

In [9]:
count = 0
end = 8
total = 0
while count < 8:
    a = int(input('Please enter a number'))
    total = total + a
    count += 1
print(total)

8


In [5]:
count = 0
for num in range(0, 8, 1):
    a = float(input('Please enter a number'))
    count = count + a
print(count) 

8.0


#### 3	
Write a Program to compute the average of ten numbers. You will prompt the user for each number.

In [ ]:
count = 0
end = 10
total = 0
while count < end:
    a = float(input('Please enter a number.'))
    total = total + a
    count += 1
total = total / 10
print(total)

45.0
4.5


#### 4
Write a program to calculate the sum of numbers from 1 to 100.

In [ ]:
# we are adding total of 99 times from 1 to 100
count = 0
end = 100
total = 0
while count < end:
    count += 1
    total = total + count
print(total)


5050


In [8]:
sum = 0
for x  in range(1, 101, 1):
    sum = sum + x
print(sum)

5050


#### 5
Write a program to print the numbers from 10 down to 1 on a single line.


In [ ]:
count = 0
end = 10
while count < end:
    count += 1
    print(count, end = (' '))
    # use end function in print for same line.
print('\nhello \nworld')
# use \n for new line.

1 2 3 4 5 6 7 8 9 10 
hello 
world


In [9]:
for x in range(1, 11, 1):
    print(x, end = (' '))

1 2 3 4 5 6 7 8 9 10 

#### 6
Write a program to prompt the user for a number and then prints its multiplication table up to 10

In [ ]:
count = 1
# is easier to make count 1 so we could multiply it by 1 directly.
end = 11
total = 0
a = float(input('Please enter a number.'))
while count < end:
    total = a * count
    count += 1
    print(total)

2.0
4.0
6.0
8.0
10.0
12.0
14.0
16.0
18.0
20.0


In [13]:
sum = float(input('Please enter a number.'))
total = 0
for x in range(1, 11, 1):
    total = sum * x
    print(total)

2.0
4.0
6.0
8.0
10.0
12.0
14.0
16.0
18.0
20.0


#### 7 
Write a program to print all the numbers from 1 tp 50 that are multiples of 3 or 5 or both.

In [ ]:
count = 1
end = 51
while count < end:
    if count % 3 == 0 or count % 5 == 0:
        print(count)
    count += 1

3
5
6
9
10
12
15
18
20
21
24
25
27
30
33
35
36
39
40
42
45
48
50


In [14]:
for x in range (1, 51, 1):
    if x % 3 == 0 or x % 5 == 0:
        print(x)

3
5
6
9
10
12
15
18
20
21
24
25
27
30
33
35
36
39
40
42
45
48
50


#### 8
Write a program to print all the numbers from 40 to 0 that are NOT multiples of 7.

In [4]:
count = 40
end = 0
while count >= end:
    if count % 7 != 0:
        print(count)
    count -= 1

40
39
38
37
36
34
33
32
31
30
29
27
26
25
24
23
22
20
19
18
17
16
15
13
12
11
10
9
8
6
5
4
3
2
1


In [15]:
for x in range(1, 41, 1):
    if x % 7 != 0:
        print(x)

1
2
3
4
5
6
8
9
10
11
12
13
15
16
17
18
19
20
22
23
24
25
26
27
29
30
31
32
33
34
36
37
38
39
40


#### 9	
Write a python program that displays a conversion table from gallons to liters. The program should display gallons from 10 to 20 in one-gallon increments and the corresponding liter equivalents. Use the relationship that 1 gallon contains 3.785 liters. Your table must look professional!

In [37]:
#from 10 to 20 is 11 entries
header = 'Gallon      Litters'
litter = 0
print(header)
print('=' * len(header))
for gallon in range(10, 21, 1):
# range function takes in first range and second and last entry is the increment
   
    litter = f'{(gallon * 3.785):.2f}'
    print(f'{gallon:<10}{litter:>7}')

Gallon      Litters
10          37.85
11          41.64
12          45.42
13          49.20
14          52.99
15          56.78
16          60.56
17          64.34
18          68.13
19          71.92
20          75.70


#### 10	
Write a C# program that converts feet to meters. The program should display feet from 3 to 30 in three-foot increments and the corresponding meter equivalents. Use the relationship that 1 meter is equivalent to 3.28 feet. Your table must look professional!

| meters |  feet |
| ---: | ---: |
|    3 |      9.75 |
|    6 |     19.50 |
|    9 |     29.25 |
|   12 |     39.00 |
|   15 |     48.75 |
|   18 |     58.50 |
|   21 |     68.25 |
|   24 |     78.00 |
|   27 |     87.75 |
|   30 |     97.50 |

In [89]:
header = 'meters       feet'
line = '|'
print(header)
print('=' * len(header))
feet = 0
for meters in range(3, 33, 3):
    feet = f'{meters * 3.28:.3f}'
    print(f'{line}{meters:>3}{line:>5}{feet:>7}{line:>1}')
print('=' * len(header))

meters       feet
|  3    |  9.840|
|  6    | 19.680|
|  9    | 29.520|
| 12    | 39.360|
| 15    | 49.200|
| 18    | 59.040|
| 21    | 68.880|
| 24    | 78.720|
| 27    | 88.560|
| 30    | 98.400|


#### 11	
Write a program to produce a table of the numbers 0 through 20 in increments of 2, with their squares and cubes. Your table must look professional!

Your output should be formatted as shown below

| x  |  x^2 |   x^3 |
| ---: | ---: | ---: |
|  0 |      0 |       0 |
|  2 |      4 |       8 |
|  4 |     16 |      64 |
|  6 |     36 |     216 |
|  8 |     64 |     512 |
| 10 |    100 |    1000 |
| 12 |    144 |    1728 |
| 14 |    196 |    2744 |
| 16 |    256 |    4096 |
| 18 |    324 |    5832 |
| 20 |    400 |    8000 |

In [42]:
x = 'x   x^2    x^3'
print(x)
print('=' * len(x))
for a in range(0, 22, 2):
    b = a**2
    c = a**3
    print(f'{a:>1} {b:>4} {c:>6,}')


x   x^2    x^3
0    0      0
2    4      8
4   16     64
6   36    216
8   64    512
10  100  1,000
12  144  1,728
14  196  2,744
16  256  4,096
18  324  5,832
20  400  8,000


#### 12	
Write a program to produce a table of numbers from 10 to 1, with their squares and cubes. Again, your table must look professional!

Your output should be formatted as shown below
| x  |  x^2 |   x^3 |
| ---: | ---: | ---: |
 | 10 |     100 |    1000 |
 |  9 |      81 |     729 |
 |  8 |      64 |     512 |
 |  7 |      49 |     343 |
 |  6 |      36 |     216 |
 |  5 |      25 |     125 |
 |  4 |      16 |      64 |
 |  3 |       9 |      27 |
 |  2 |       4 |       8 |
 |  1 |       1 |       1 |

In [104]:
header = 'x      x^2      x^3'
print(header)
print('=' * len(header))
for x in range(10, 0, -1):
    print(f'{x:>2}{x**2:>8}{x**3:>8}')

x      x^2      x^3
10     100    1000
 9      81     729
 8      64     512
 7      49     343
 6      36     216
 5      25     125
 4      16      64
 3       9      27
 2       4       8
 1       1       1


#### 13	
Write a program that displays a conversion table from Fahrenheit to Celsius. The program must request the starting Fahrenheit value, the ending Fahrenheit value, and the increment. Thus, instead of the condition checking for a fixed count, the condition checks for the ending Fahrenheit value. Your table must look professional!

Your output should be formatted as shown below
|meters |  feet|
| ---: | ---: |
|    0 |      0.00 |
|    3 |      9.75 |
|    6 |     19.50 |
|    9 |     29.25 |
|   12 |     39.00 |
|   15 |     48.75 |
|   18 |     58.50 |
|   21 |     68.25 |
|   24 |     78.00 |
|   27 |     87.75 |
|   30 |     97.50 |

In [80]:
fehren1 = float(input('Please input the first Fehrenheit(biggest).'))
fehren2 = float(input('Please input the second Fehrenheit(Smallest).'))
inc = float(input('Please input the inncrement.'))
diff = fehren1 - fehren2
many = int(diff/inc) + 1
a = 'Fehrenheit         Celcius          Increment'
print(a)
print('=' * len(a))
for x in range(0, many, 1):
    cel = (fehren1 - inc * x - 32) / 1.8
    print(f'{fehren1 - x * inc:>2.4f} {cel:>19.4f} {inc:>15}')


Fehrenheit         Celcius          Increment
342.6000            172.5556             2.7
339.9000            171.0556             2.7
337.2000            169.5556             2.7
334.5000            168.0556             2.7
331.8000            166.5556             2.7
329.1000            165.0556             2.7
326.4000            163.5556             2.7
323.7000            162.0556             2.7
321.0000            160.5556             2.7
318.3000            159.0556             2.7
315.6000            157.5556             2.7
312.9000            156.0556             2.7
310.2000            154.5556             2.7
307.5000            153.0556             2.7
304.8000            151.5556             2.7
302.1000            150.0556             2.7
299.4000            148.5556             2.7
296.7000            147.0556             2.7
294.0000            145.5556             2.7
291.3000            144.0556             2.7
288.6000            142.5556             2.7
285.9000 

#### 14  
Write a python program that calculates and displays the amount of money available in a bank account that initially has $1,000 deposited in it and that earns 8 percent interest a year. Your program should display the amount available at the end of each year for a period of ten years. Use the relationship that the money available at the end of each year equals the amount of money in the account at the start of the year plus .08 times the amount available at the start of the year.


Your output should be formatted as shown below
| Year  |  Balance |
| ---: | ---: |
|  1 |     $1,080.00 |
|  2 |     $1,166.40 |
|  3 |     $1,259.71 |
|  4 |     $1,360.49 |
|  5 |     $1,469.33 |
|  6 |     $1,586.87 |
|  7 |     $1,713.82 |
|  8 |     $1,850.93 |
|  9 |     $1,999.00 |
| 10 |     $2,158.92 |


In [23]:
a = 'year     Balance '
print(a)
print('=' * len(a))
initial = 1000
for x in range(1, 11, 1):
    initial = initial * (1.08)
    print(f'{x:>3} {initial:>12.2f}')

year     Balance 
  1      1080.00
  2      1166.40
  3      1259.71
  4      1360.49
  5      1469.33
  6      1586.87
  7      1713.82
  8      1850.93
  9      1999.00
 10      2158.92


#### 15
Write a program to prompt the user for an integer. The program should add the digits and print the sum.

In [ ]:
a = int(input('Please enter integers'))

#### 16	
Write a program that continuously requests a grade. 
If the grade is less than 0 or greater than 100, your program should print an appropriate message informing the user that an invalid grade has been entered, 
else the grade should be added to a total. 
If the grade is 999, the program should exit the loop and display the sum, number of valid grades and the average of the valid grades entered. (You must NOT display an invalid message)
P.S. The value 999 is use to terminate the cycle and should not be used in any of the calculation. You must not display an “Invalid grade”

In [ ]:
grade = 0
for x in range(0, 999, 1):
    a = int(input('Please enter grade.'))
    if a < 0 or a > 100:
        print('Invalid grade')
        continue
    if grade > 999:
        print(grade)
        break
    grade = grade + a

1000
1000


#### 17	
Print the decimal, octal, and hexadecimal values of all characters between the start and stop characters entered by a user. For example, if the user enters an a and a z, the program should print all the characters between a and z and their respective numerical values. Make sure that the second character entered by the user occurs later in the alphabet than the first character. If it does not, write a loop that repeatedly asks the user for a valid second character until one is entered.
Your output should be formatted as shown below
| Letter | Decimal | Octal | Hex |
|--------|---------|-------|-----|
| a      | 97	   | 141   | 61  |
| b      | 98	   | 142   | 62  |
| c      | 99	   | 143   | 63  |
| d      | 100     | 144   | 64  |
| e      | 101     | 145   | 65  |

-   to obtain an int from a char use an explicit cast
-   to obtain an octal from an int use Convert.ToString(«number_value», 8)
-   to obtain a hexadecimal from an int use Convert.ToString(«number_value», 16)

In [ ]:
char = 'a'
print(ord(char))
a = input('Please enter the first character.')
for x in(0, 999, 1):
    b = input('Please enter the second character.')
    if ord(a) <= ord(b):
        break
    print('Please enter a character after the first character')
print(f'{a}{b}')
title = 'Letter     Decimal       Octal        Hex'
print(title)
print('=' * len(title))
int_a = ord(a)
int_b = ord(b)

for y in range(int_a, int_b+1, 1):
    print(f'{chr(y):>4} {y:>10} {oct(y)[2:]:>12}{hex(y)[-2:]:>12}')
print(f'{int_a} {int_b}')

97
Please enter a character after the first character
Please enter a character after the first character
z|
Letter     Decimal       Octal        Hex
   z        122          172          7a
   {        123          173          7b
   |        124          174          7c
122 124


#### 18
Write a program to print all the letter from A to Z

In [44]:

for i in range (97, 123, 1):
    print(chr(i), end=' ')

a b c d e f g h i j k l m n o p q r s t u v w x y z 

#### 19
Write a program to find al 3-digits Armstrong number (e.g. 153 = 1^3 + 5^3 + 3^3)

In [ ]:
#Looping through 100 to 999
for num in range(100, 1000, 1):
    #making int into string
    char_num = str(num)
    #Declaring total = 0 at the first loop since we want to clear it after we 
    #have done with comparing total and num.
    total = 0
    #Creating a loop within a loop so we can loop though each character of str(num)
    for char in char_num:
        #Each digit is retrieved by looking at each character of
        #str(num) and making it into a integer
        digit = int(char)
        #Adding all the digit to see if num is a armstrong number
        total += digit ** 3
        #test if num is equal to total, if yes print it
        if total == num:
            #remember if u want the first character of the strong it starts from 0.
            print(f'{num} = {char_num[0]}^3 + {char_num[1]}^3 + {char_num[2]}^3')
        


153 = 1^3 + 5^3 + 3^3
370 = 3^3 + 7^3 + 0^3
370 = 3^3 + 7^3 + 0^3
371 = 3^3 + 7^3 + 1^3
407 = 4^3 + 0^3 + 7^3


In [ ]:
print("3-digit Armstrong numbers are:")
#Loop should be inside 100 and 999.
for num in range(100, 1000):
    #This is list comprehension
    digits = [int(d) for d in str(num)]
    if sum(d**3 for d in digits) == num:
        print(f"{num} = {' + '.join(f'{d}³' for d in digits)}")




3-digit Armstrong numbers are:
153 = 1³ + 5³ + 3³
370 = 3³ + 7³ + 0³
371 = 3³ + 7³ + 1³
407 = 4³ + 0³ + 7³
